### DISCUSSION TITLE
Error during instalation with regards to email

### DISCUSSION THREAD

**Discussion by PrinceDeveloperOf** [firefly-iii/firefly-iii#7118] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I run `sudo composer create-project grumpydictator/firefly-iii --no-dev --ignore-platform-req=php` to install the package it mostly works except for one error telling me an email must have "To", "CC", or "Bcc" header

### Expected behaviour

I expect it to install without this error.

### Debug information

`
```
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? 
Creating a "grumpydictator/firefly-iii" project at "./firefly-iii"
Info from https://repo.packagist.org: #StandWithUkraine
Installing grumpydictator/firefly-iii (v6.0.0)
  - Installing grumpydictator/firefly-iii (v6.0.0): Extracting archive
Created project in /var/www/html/firefly-iii
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Package operations: 114 installs, 0 updates, 0 removals
  - Installing dasprid/enum (1.0.4): Extracting archive
  - Installing voku/portable-ascii (2.0.1): Extracting archive
  - Installing symfony/polyfill-php80 (v1.27.0): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.27.0): Extracting archive
  - Installing symfony/polyfill-ctype (v1.27.0): Extracting archive
  - Installing phpoption/phpoption (1.9.1): Extracting archive
  - Installing graham-campbell/result-type (v1.1.1): Extracting archive
  - Installing vlucas/phpdotenv (v5.5.0): Extracting archive
  - Installing symfony/css-selector (v6.2.7): Extracting archive
  - Installing tijsverkoyen/css-to-inline-styles (2.2.6): Extracting archive
  - Installing symfony/var-dumper (v6.2.7): Extracting archive
  - Installing symfony/polyfill-uuid (v1.27.0): Extracting archive
  - Installing symfony/uid (v6.2.7): Extracting archive
  - Installing symfony/routing (v6.2.7): Extracting archive
  - Installing symfony/process (v6.2.7): Extracting archive
  - Installing symfony/polyfill-php72 (v1.27.0): Extracting archive
  - Installing symfony/polyfill-intl-normalizer (v1.27.0): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.27.0): Extracting archive
  - Installing symfony/mime (v6.2.7): Extracting archive
  - Installing psr/container (2.0.2): Extracting archive
  - Installing symfony/service-contracts (v3.2.1): Extracting archive
  - Installing psr/event-dispatcher (1.0.0): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v3.2.1): Extracting archive
  - Installing symfony/event-dispatcher (v6.2.7): Extracting archive
  - Installing psr/log (3.0.0): Extracting archive
  - Installing doctrine/lexer (3.0.0): Extracting archive
  - Installing egulias/email-validator (4.0.1): Extracting archive
  - Installing symfony/mailer (v6.2.7): Extracting archive
  - Installing symfony/deprecation-contracts (v3.2.1): Extracting archive
  - Installing symfony/http-foundation (v6.2.7): Extracting archive
  - Installing symfony/error-handler (v6.2.7): Extracting archive
  - Installing symfony/http-kernel (v6.2.7): Extracting archive
  - Installing symfony/finder (v6.2.7): Extracting archive
  - Installing symfony/polyfill-intl-grapheme (v1.27.0): Extracting archive
  - Installing symfony/string (v6.2.7): Extracting archive
  - Installing symfony/console (v6.2.7): Extracting archive
  - Installing ramsey/collection (2.0.0): Extracting archive
  - Installing brick/math (0.10.2): Extracting archive
  - Installing ramsey/uuid (4.7.3): Extracting archive
  - Installing psr/simple-cache (3.0.0): Extracting archive
  - Installing nunomaduro/termwind (v1.15.1): Extracting archive
  - Installing symfony/translation-contracts (v3.2.1): Extracting archive
  - Installing symfony/translation (v6.2.7): Extracting archive
  - Installing nesbot/carbon (2.66.0): Extracting archive
  - Installing monolog/monolog (3.3.1): Extracting archive
  - Installing league/mime-type-detection (1.11.0): Extracting archive
  - Installing league/flysystem (3.12.3): Extracting archive
  - Installing nette/utils (v4.0.0): Extracting archive
  - Installing nette/schema (v1.2.3): Extracting archive
  - Installing dflydev/dot-access-data (v3.0.2): Extracting archive
  - Installing league/config (v1.2.0): Extracting archive
  - Installing league/commonmark (2.3.9): Extracting archive
  - Installing laravel/serializable-closure (v1.3.0): Extracting archive
  - Installing guzzlehttp/uri-template (v1.0.1): Extracting archive
  - Installing fruitcake/php-cors (v1.2.0): Extracting archive
  - Installing webmozart/assert (1.11.0): Extracting archive
  - Installing dragonmantank/cron-expression (v3.3.2): Extracting archive
  - Installing doctrine/inflector (2.0.6): Extracting archive
  - Installing laravel/framework (v10.2.0): Extracting archive
  - Installing facade/ignition-contracts (1.0.2): Extracting archive
  - Installing diglactic/laravel-breadcrumbs (v8.1.0): Extracting archive
  - Installing psr/cache (3.0.0): Extracting archive
  - Installing doctrine/event-manager (2.0.0): Extracting archive
  - Installing doctrine/deprecations (v1.0.0): Extracting archive
  - Installing doctrine/cache (2.2.0): Extracting archive
  - Installing doctrine/dbal (3.6.1): Extracting archive
  - Installing gdbots/query-parser (v3.0.0): Extracting archive
  - Installing guzzlehttp/promises (1.5.2): Extracting archive
  - Installing ralouphie/getallheaders (3.0.3): Extracting archive
  - Installing psr/http-message (1.0.1): Extracting archive
  - Installing psr/http-factory (1.0.1): Extracting archive
  - Installing guzzlehttp/psr7 (2.4.3): Extracting archive
  - Installing paragonie/constant_time_encoding (v2.6.3): Extracting archive
  - Installing pragmarx/google2fa (v8.0.1): Extracting archive
  - Installing bacon/bacon-qr-code (2.0.8): Extracting archive
  - Installing pragmarx/google2fa-qrcode (v1.0.3): Extracting archive
  - Installing jc5/google2fa-laravel (v2.0.8): Extracting archive
  - Installing pragmarx/random (v0.2.2): Extracting archive
  - Installing jc5/recovery (v2.2.0): Extracting archive
  - Installing symfony/psr-http-message-bridge (v2.1.4): Extracting archive
  - Installing paragonie/random_compat (v9.99.100): Extracting archive
  - Installing phpseclib/phpseclib (3.0.18): Extracting archive
  - Installing php-http/message-factory (v1.0.2): Extracting archive
  - Installing nyholm/psr7 (1.5.1): Extracting archive
  - Installing league/uri-interfaces (2.3.0): Extracting archive
  - Installing league/uri (6.8.0): Extracting archive
  - Installing league/event (2.2.0): Extracting archive
  - Installing psr/clock (1.0.0): Extracting archive
  - Installing lcobucci/clock (3.0.0): Extracting archive
  - Installing lcobucci/jwt (4.3.0): Extracting archive
  - Installing defuse/php-encryption (v2.3.1): Extracting archive
  - Installing league/oauth2-server (8.4.0): Extracting archive
  - Installing firebase/php-jwt (v6.4.0): Extracting archive
  - Installing laravel/passport (v11.8.3): Extracting archive
  - Installing laravel/sanctum (v3.2.1): Extracting archive
  - Installing psr/http-client (1.0.1): Extracting archive
  - Installing guzzlehttp/guzzle (7.5.0): Extracting archive
  - Installing laravel/slack-notification-channel (v2.5.0): Extracting archive
  - Installing laravel/ui (v4.2.1): Extracting archive
  - Installing laravelcollective/html (v6.4.0): Extracting archive
  - Installing league/csv (9.8.0): Extracting archive
  - Installing league/fractal (0.20.1): Extracting archive
  - Installing filp/whoops (2.14.6): Extracting archive
  - Installing nunomaduro/collision (v7.0.5): Extracting archive
  - Installing predis/predis (v2.1.2): Extracting archive
  - Installing twig/twig (v3.5.1): Extracting archive
  - Installing rcrowe/twigbridge (v0.14.1): Extracting archive
  - Installing spatie/backtrace (1.2.2): Extracting archive
  - Installing spatie/flare-client-php (1.3.5): Extracting archive
  - Installing spatie/ignition (1.4.5): Extracting archive
  - Installing spatie/laravel-ignition (2.0.0): Extracting archive
  - Installing symfony/http-client-contracts (v3.2.1): Extracting archive
  - Installing symfony/http-client (v6.2.7): Extracting archive
  - Installing symfony/mailgun-mailer (v6.2.7): Extracting archive
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
73 packages you are using are looking for funding.
Use the \`composer fund\` command to find out more!
> @php artisan firefly:instructions install
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v6.0.0!                                |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
> @php artisan firefly-iii:verify-security-alerts
[2023-03-04 08:16:04] local.ERROR: Exception is: {"class":"Illuminate\\Database\\QueryException","errorMessage":"SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for db failed: Temporary failure in name resolution (Connection: mysql, SQL: delete from `configuration` where `name` = upgrade_security_message)","time":"Sat, 04 Mar 2023 08:16:04 +0100","file":"\/var\/www\/html\/firefly-iii\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php","line":760,"code":2002,"version":"6.0.0","url":"http:\/\/localhost","userAgent":"Symfony","json":true,"method":"GET"}  
[2023-03-04 08:16:04] local.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"An email must have a \"To\", \"Cc\", or \"Bcc\" header.","time":"Sat, 04 Mar 2023 08:16:04 +0100","file":"\/var\/www\/html\/firefly-iii\/app\/Jobs\/MailError.php","line":94,"code":0,"version":"6.0.0","url":"http:\/\/localhost","userAgent":"Symfony","json":true,"method":"GET"}  

   FireflyIII\Exceptions\FireflyException 

  An email must have a "To", "Cc", or "Bcc" header.

  at app/Jobs/MailError.php:94
     90▕                         }
     91▕                     }
     92▕                 );
     93▕             } catch (Exception $e) { // intentional generic exception
  ➜  94▕                 throw new FireflyException($e->getMessage(), 0, $e);
     95▕             }
     96▕         }
     97▕     }
     98▕ }

      +8 vendor frames 

  9   app/Jobs/MailError.php:84
      Illuminate\Support\Facades\Facade::__callStatic()
      +21 vendor frames 
  31  app/Exceptions/Handler.php:223

No security vulnerability advisories found
> @php artisan key:generate

   INFO  Application key set successfully.  
`

```
I'm running Xubuntu 20.04

### Additional info

I use the flag `--ignore-platform-req=php` because if I don't it tells me that php 8.2 won't work for some dependencies.

**Comment by JC5** [comment, marked as answer]
I see a database error and then a message about emails (it tries to email the error if you have configured it as such).

**Reply by varioous** [reply]
Same here

` 
php composer.phar create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii-updated 6.0
Creating a "grumpydictator/firefly-iii" project at "./firefly-iii-updated"

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/www/users/finank/composer.phar/src/Composer/Autoload/AutoloadGenerator.php on line 879

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in phar:///usr/www/users/finank/composer.phar/src/Composer/Autoload/AutoloadGenerator.php on line 884
Info from https://repo.packagist.org: #StandWithUkraine
Installing grumpydictator/firefly-iii (v6.0.0)
  - Installing grumpydictator/firefly-iii (v6.0.0): Extracting archive
Created project in /usr/www/users/finank/firefly-iii-updated
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Package operations: 114 installs, 0 updates, 0 removals
  - Installing dasprid/enum (1.0.4): Extracting archive
  - Installing voku/portable-ascii (2.0.1): Extracting archive
  - Installing symfony/polyfill-php80 (v1.27.0): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.27.0): Extracting archive
  - Installing symfony/polyfill-ctype (v1.27.0): Extracting archive
  - Installing phpoption/phpoption (1.9.1): Extracting archive
  - Installing graham-campbell/result-type (v1.1.1): Extracting archive
  - Installing vlucas/phpdotenv (v5.5.0): Extracting archive
  - Installing symfony/css-selector (v6.2.7): Extracting archive
  - Installing tijsverkoyen/css-to-inline-styles (2.2.6): Extracting archive
  - Installing symfony/var-dumper (v6.2.7): Extracting archive
  - Installing symfony/polyfill-uuid (v1.27.0): Extracting archive
  - Installing symfony/uid (v6.2.7): Extracting archive
  - Installing symfony/routing (v6.2.7): Extracting archive
  - Installing symfony/process (v6.2.7): Extracting archive
  - Installing symfony/polyfill-php72 (v1.27.0): Extracting archive
  - Installing symfony/polyfill-intl-normalizer (v1.27.0): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.27.0): Extracting archive
  - Installing symfony/mime (v6.2.7): Extracting archive
  - Installing psr/container (2.0.2): Extracting archive
  - Installing symfony/service-contracts (v3.2.1): Extracting archive
  - Installing psr/event-dispatcher (1.0.0): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v3.2.1): Extracting archive
  - Installing symfony/event-dispatcher (v6.2.7): Extracting archive
  - Installing psr/log (3.0.0): Extracting archive
  - Installing doctrine/lexer (3.0.0): Extracting archive
  - Installing egulias/email-validator (4.0.1): Extracting archive
  - Installing symfony/mailer (v6.2.7): Extracting archive
  - Installing symfony/deprecation-contracts (v3.2.1): Extracting archive
  - Installing symfony/http-foundation (v6.2.7): Extracting archive
  - Installing symfony/error-handler (v6.2.7): Extracting archive
  - Installing symfony/http-kernel (v6.2.7): Extracting archive
  - Installing symfony/finder (v6.2.7): Extracting archive
  - Installing symfony/polyfill-intl-grapheme (v1.27.0): Extracting archive
  - Installing symfony/string (v6.2.7): Extracting archive
  - Installing symfony/console (v6.2.7): Extracting archive
  - Installing ramsey/collection (2.0.0): Extracting archive
  - Installing brick/math (0.10.2): Extracting archive
  - Installing ramsey/uuid (4.7.3): Extracting archive
  - Installing psr/simple-cache (3.0.0): Extracting archive
  - Installing nunomaduro/termwind (v1.15.1): Extracting archive
  - Installing symfony/translation-contracts (v3.2.1): Extracting archive
  - Installing symfony/translation (v6.2.7): Extracting archive
  - Installing nesbot/carbon (2.66.0): Extracting archive
  - Installing monolog/monolog (3.3.1): Extracting archive
  - Installing league/mime-type-detection (1.11.0): Extracting archive
  - Installing league/flysystem (3.12.3): Extracting archive
  - Installing nette/utils (v4.0.0): Extracting archive
  - Installing nette/schema (v1.2.3): Extracting archive
  - Installing dflydev/dot-access-data (v3.0.2): Extracting archive
  - Installing league/config (v1.2.0): Extracting archive
  - Installing league/commonmark (2.3.9): Extracting archive
  - Installing laravel/serializable-closure (v1.3.0): Extracting archive
  - Installing guzzlehttp/uri-template (v1.0.1): Extracting archive
  - Installing fruitcake/php-cors (v1.2.0): Extracting archive
  - Installing webmozart/assert (1.11.0): Extracting archive
  - Installing dragonmantank/cron-expression (v3.3.2): Extracting archive
  - Installing doctrine/inflector (2.0.6): Extracting archive
  - Installing laravel/framework (v10.2.0): Extracting archive
  - Installing facade/ignition-contracts (1.0.2): Extracting archive
  - Installing diglactic/laravel-breadcrumbs (v8.1.0): Extracting archive
  - Installing psr/cache (3.0.0): Extracting archive
  - Installing doctrine/event-manager (2.0.0): Extracting archive
  - Installing doctrine/deprecations (v1.0.0): Extracting archive
  - Installing doctrine/cache (2.2.0): Extracting archive
  - Installing doctrine/dbal (3.6.1): Extracting archive
  - Installing gdbots/query-parser (v3.0.0): Extracting archive
  - Installing guzzlehttp/promises (1.5.2): Extracting archive
  - Installing ralouphie/getallheaders (3.0.3): Extracting archive
  - Installing psr/http-message (1.0.1): Extracting archive
  - Installing psr/http-factory (1.0.1): Extracting archive
  - Installing guzzlehttp/psr7 (2.4.3): Extracting archive
  - Installing paragonie/constant_time_encoding (v2.6.3): Extracting archive
  - Installing pragmarx/google2fa (v8.0.1): Extracting archive
  - Installing bacon/bacon-qr-code (2.0.8): Extracting archive
  - Installing pragmarx/google2fa-qrcode (v1.0.3): Extracting archive
  - Installing jc5/google2fa-laravel (v2.0.8): Extracting archive
  - Installing pragmarx/random (v0.2.2): Extracting archive
  - Installing jc5/recovery (v2.2.0): Extracting archive
  - Installing symfony/psr-http-message-bridge (v2.1.4): Extracting archive
  - Installing paragonie/random_compat (v9.99.100): Extracting archive
  - Installing phpseclib/phpseclib (3.0.18): Extracting archive
  - Installing php-http/message-factory (v1.0.2): Extracting archive
  - Installing nyholm/psr7 (1.5.1): Extracting archive
  - Installing league/uri-interfaces (2.3.0): Extracting archive
  - Installing league/uri (6.8.0): Extracting archive
  - Installing league/event (2.2.0): Extracting archive
  - Installing psr/clock (1.0.0): Extracting archive
  - Installing lcobucci/clock (3.0.0): Extracting archive
  - Installing lcobucci/jwt (4.3.0): Extracting archive
  - Installing defuse/php-encryption (v2.3.1): Extracting archive
  - Installing league/oauth2-server (8.4.0): Extracting archive
  - Installing firebase/php-jwt (v6.4.0): Extracting archive
  - Installing laravel/passport (v11.8.3): Extracting archive
  - Installing laravel/sanctum (v3.2.1): Extracting archive
  - Installing psr/http-client (1.0.1): Extracting archive
  - Installing guzzlehttp/guzzle (7.5.0): Extracting archive
  - Installing laravel/slack-notification-channel (v2.5.0): Extracting archive
  - Installing laravel/ui (v4.2.1): Extracting archive
  - Installing laravelcollective/html (v6.4.0): Extracting archive
  - Installing league/csv (9.8.0): Extracting archive
  - Installing league/fractal (0.20.1): Extracting archive
  - Installing filp/whoops (2.14.6): Extracting archive
  - Installing nunomaduro/collision (v7.0.5): Extracting archive
  - Installing predis/predis (v2.1.2): Extracting archive
  - Installing twig/twig (v3.5.1): Extracting archive
  - Installing rcrowe/twigbridge (v0.14.1): Extracting archive
  - Installing spatie/backtrace (1.2.2): Extracting archive
  - Installing spatie/flare-client-php (1.3.5): Extracting archive
  - Installing spatie/ignition (1.4.5): Extracting archive
  - Installing spatie/laravel-ignition (2.0.0): Extracting archive
  - Installing symfony/http-client-contracts (v3.2.1): Extracting archive
  - Installing symfony/http-client (v6.2.7): Extracting archive
  - Installing symfony/mailgun-mailer (v6.2.7): Extracting archive
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
73 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> @php artisan firefly:instructions install
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v6.0.0!                                |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
> @php artisan firefly-iii:verify-security-alerts
[2023-03-04 13:38:43] local.ERROR: Exception is: {"class":"Illuminate\\Database\\QueryException","errorMessage":"SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for db failed: Name or service not known (Connection: mysql, SQL: delete from `configuration` where `name` = upgrade_security_message)","time":"Sat, 04 Mar 2023 13:38:43 +0100","file":"\/usr\/www\/users\/finank\/firefly-iii-updated\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php","line":760,"code":2002,"version":"6.0.0","url":"http:\/\/localhost","userAgent":"Symfony","json":true,"method":"GET"}
[2023-03-04 13:38:43] local.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"An email must have a \"To\", \"Cc\", or \"Bcc\" header.","time":"Sat, 04 Mar 2023 13:38:43 +0100","file":"\/usr\/www\/users\/finank\/firefly-iii-updated\/app\/Jobs\/MailError.php","line":94,"code":0,"version":"6.0.0","url":"http:\/\/localhost","userAgent":"Symfony","json":true,"method":"GET"}

   FireflyIII\Exceptions\FireflyException

  An email must have a "To", "Cc", or "Bcc" header.

  at app/Jobs/MailError.php:94
     90▕                         }
     91▕                     }
     92▕                 );
     93▕             } catch (Exception $e) { // intentional generic exception
  ➜  94▕                 throw new FireflyException($e->getMessage(), 0, $e);
     95▕             }
     96▕         }
     97▕     }
     98▕ }

      +8 vendor frames

  9   app/Jobs/MailError.php:84
      Illuminate\Support\Facades\Facade::__callStatic()
      +21 vendor frames
  31  app/Exceptions/Handler.php:223

> @php artisan key:generate

   INFO  Application key set successfully.
`

**Reply by PrinceDeveloperOf** [reply]
I had negleted to make a table and user to access it. 

After realizing that i made a user and a table in mysql and changed the `.env` file. 

This error happened during the installation am i supposed to change the .env file before using composer?

**Reply by JC5** [reply]
I'm not sure I understand what you mean. You have a database server called "db" now?

**Reply by PrinceDeveloperOf** [reply]
No i have a mysql database called firefly. I put the information in the .env file and did the rest of tutorial; successfully setting up an instance of firefly.

But if i deleted that instance of firefly and ran composer again it would still give me the database error again i think, because the .env file hasn't been configured.
